Update Resonance_Compensation.md to recommend SET_VELOCITY_LIMIT

KevinOConnor opened the pull request:
I noticed that Resonance_Compensation.md was still recommending updating the printer.cfg file to set max_accel=7000.  It's simpler to make all those adjustments by issuing a `SET_VELOCITY_LIMIT` command.  This PR updates the document.

@dmbutyugin - FYI.

-Kevin

dmbutyugin commented on docs/Resonance_Compensation.md:
```diff
@@ -79,14 +72,14 @@ First, measure the **ringing frequency**.
+   where *V* is the velocity for outer perimeters (mm/sec). For the example
+   above, we marked 6 oscillations, and the test was printed at 100 mm/sec
+   velocity, so the frequency is 100 * 6 / 12.14 ≈ 49.4 Hz.
+10. Do (9) - (11) for Y mark as well.
```
This should read 'Do (7) - (9) for Y mark as well.' now (or different numbers, if the square_corner_velocity point is kept).

dmbutyugin commented on docs/Resonance_Compensation.md:
```diff
@@ -45,31 +44,25 @@ Slice the ringing test model, which can be found in
-6. Execute the command
-   `TUNING_TOWER COMMAND=SET_VELOCITY_LIMIT PARAMETER=ACCEL START=1250 FACTOR=100 BAND=5`.
+4. Execute the command:
+   `TUNING_TOWER COMMAND=SET_VELOCITY_LIMIT PARAMETER=ACCEL START=1250 FACTOR=100 BAND=5`
```
I see that TUNING_TOWER has now the STEP_DELTA and STEP_HEIGHT parameters. Would it make sense to update this command here and below in other instances to use them instead? I think we get countless questions on discord where people are confused and ask if the command is wrong. It will read

`TUNING_TOWER COMMAND=SET_VELOCITY_LIMIT PARAMETER=ACCEL START=1500 STEP_DELTA=100 STEP_HEIGHT=5`

if I am not mistaken

dmbutyugin commented on docs/Resonance_Compensation.md:
```diff
@@ -45,31 +44,25 @@ Slice the ringing test model, which can be found in
-   `printer.cfg` to 7000. Note that this is only needed for tuning, and more
-   proper value will be selected in the corresponding
-   [section](#selecting-max_accel).
-2. If `square_corner_velocity` parameter was changed, revert it back to 5.0.
```
Note that the way instructions go, it was suggested to set square_corner_velocity back to 5.0 and not change it back after tuning. If we remove this sentence, and instead suggest to set it via `SET_VELOCITY_LIMIT SQUARE_CORNER_VELOCITY=5`, then the testing will be done with one value, but the real printing will be done with a different value. This may be undesirable. So, maybe it makes sense to keep this point? Or mention it earlier, in the non-numbered part of the section. Alternatively, we should probably not suggest to reset SQUARE_CORNER_VELOCITY. However my estimations (via calibrate_shaper.py) indicate that having higher SQUARE_CORNER_VELOCITY very quickly increases the smoothing, such that SQUARE_CORNER_VELOCITY==10 will already create too much smoothing even with rather high resonance frequencies and low accelerations, this is why the suggestion was to keep SQUARE_CORNER_VELOCITY low.

dmbutyugin commented on docs/Resonance_Compensation.md:
```diff
@@ -150,16 +143,16 @@ For most of the printers, either MZV or EI shapers can be recommended. This
-5. Print the test model sliced with the suggested parameters.
+1. Restart the firmware: `RESTART`
+2. Prepare for test:
+   `SET_VELOCITY_LIMIT ACCEL_TO_DECEL=7000 SQUARE_CORNER_VELOCITY=5`
```
Following the doc, this printed model will also be used later to choose max_accel. Therefore, I would suggest not to override SQUARE_CORNER_VELOCITY at least for this model (and maybe, not anywhere in the testing via this command, see my other comment above), as the results will be different from the real-life printing.

dmbutyugin commented on docs/Resonance_Compensation.md:
```diff
@@ -335,26 +324,26 @@ accelerometer and measure the resonances with it (refer to the
-2. `SET_PRESSURE_ADVANCE ADVANCE=0`.
-3. `SET_INPUT_SHAPER SHAPER_TYPE=2HUMP_EI SHAPER_FREQ_X=60 SHAPER_FREQ_Y=60`.
-4. `TUNING_TOWER COMMAND=SET_VELOCITY_LIMIT PARAMETER=ACCEL START=1250 FACTOR=100 BAND=5`.
+2. `SET_VELOCITY_LIMIT ACCEL_TO_DECEL=7000 SQUARE_CORNER_VELOCITY=5`
```
Actually, here too: it is better to not override SQUARE_CORNER_VELOCITY here.

dmbutyugin reviewed (commented):
Thank you for doing this, Kevin! Now indeed there is no need to change acceleration parameters via the config. I only had a few comments about the new tuning process.

KevinOConnor commented:
Thanks.

I reverted the square_corner_velocity changes.  I left in the original suggestion to not set it above 5.0 and the various tests no longer reference it.

I updated the TUNING_TOWER commands to use STEP_HEIGHT/STEP_DELTA.  BTW, I think it should be `STEP_DELTA=500`.  There is still one TUNING_TOWER command using FACTOR - I wasn't quite sure how that one is supposed to be set.

I found a few places using list numbering and attempted to fix them up.

Cheers,
-Kevin

dmbutyugin commented:
Thank you, Kevin!

> I updated the TUNING_TOWER commands to use STEP_HEIGHT/STEP_DELTA. BTW, I think it should be `STEP_DELTA=500`.

Yes, right, my bad.

> There is still one TUNING_TOWER command using FACTOR - I wasn't quite sure how that one is supposed to be set.

Yes, that one is in 'Fine-tuning resonance frequencies' section. There are explicit formulas to calculate the parameters there and how to interpret the results, so I think that one is fine. And TBH, that section was written before adxl345 support for resonance testing was added. Nowadays, I doubt that method is useful - it feels more like a waste of time. Perhaps we should delete that section eventually.

> I found a few places using list numbering and attempted to fix them up.

Yep, I didn't notice any other numbering problems either.

KevinOConnor commented:
Thanks for reviewing.  I committed this change.

> Perhaps we should delete that section eventually.

Yeah - the adxl345 mechanism is generally easier and more robust.  At some point it might make sense to simplify Resonance_Compensation.md to only have a simple calibration guide and point users to the adxl345.

-Kevin
